toolkit/tools/pkg/imagecustomizerlib/customizetdnfsnapshot.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+func setSnapshotTimeInTdnfConfig(imageChroot *safechroot.Chroot, snapshotTime imagecustomizerapi.PackageSnapshotTime) error {
+	if snapshotTime == "" {
+		return nil
+	}
+
+	// Parse to time.Time and convert to epoch
+	layout := "2006:01:02"
+	date, err := time.Parse(layout, string(snapshotTime))
+	if err != nil {
+		return fmt.Errorf("failed to parse snapshot time: %w", err)
+	}
+
+	// Use end of day to include all packages published on that date
+	date = date.Add(23*time.Hour + 59*time.Minute + 59*time.Second)
+	epoch := strconv.FormatInt(date.Unix(), 10)
+
+	tdnfConfPath := filepath.Join(imageChroot.RootDir(), "etc/tdnf/tdnf.conf")
```

Comment:
There is a question on whether the snapshot time should only be applied during customization or if it should be added to the image (which is what you are currently doing).

`--rpm-source` is only applied during customization. So, it might make sense to be consistent with that.

Note: You can pass `--config` to `tdnf` to override the location of the `tdnf.conf` file. This can be used to point tdnf at a temporary config instead of saving/restoring the actual `tdnf.conf` file.